warp10/src/main/java/io/warp10/script/functions/PERMUTE.java

```diff
@@ -0,0 +1,92 @@
+package io.warp10.script.functions;
+
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.formatted.FormattedWarpScriptFunction;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class PERMUTE extends FormattedWarpScriptFunction {
+
+  public static final String TENSOR = "tensor";
+  public static final String PATTERN = "pattern";
+  public static final String FAST = "fast";
+
+  private final Arguments args;
+  private final Arguments output;
+
+  protected Arguments getArguments() {
+    return args;
+  }
+
+  protected Arguments getOutput() {
+    return output;
+  }
+
+  public PERMUTE(String name) {
+    super(name);
+
+    getDocstring().append("Permute the dimensions of a nested LIST as if it were a tensor or a multidimensional array.");
+
+    args = new ArgumentsBuilder()
+      .addArgument(List.class, TENSOR, "The nested LIST for which dimensions will be permuted as if it were a tensor.")
+      .addListArgument(Long.class, PATTERN, "The permutation pattern (a LIST of LONG).")
+      .addOptionalArgument(Boolean.class, FAST, "If True, it does not check if the sizes of the nested lists are coherent before operating. Default to False.", false)
+      .build();
+
+    output = new ArgumentsBuilder()
+      .addArgument(List.class, TENSOR, "The resulting nested LIST.")
+      .build();
+  }
+
+  private List<Long> pattern;
```

Comment:
This should not be a field, as the apply method might be called from multiple threads